### PR TITLE
[Refactor] Move tuple desc  and add init chunk method to DataSource

### LINKED ISSUE #6003: Mem usage has exceed the limit of single query.
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
```
create external table test_export_mysql_table (
         `id_tinyint` tinyint NULL ,
         `id_smallint` smallint NULL ,
         `id_int` int NULL ,
         `id_bigint` bigint NULL ,
         `id_float` float NULL ,
         `id_double` double NULL ,
         `id_decimal` decimal(27,9) NULL ,
         `id_date` date NULL ,
         `id_datetime` datetime NULL ,
         `id_char` char(20) NULL ,
         `id_varchar` varchar(20) NULL)
         engine=MYSQL
         properties (
         "host" = "172.26.92.139",
         "port" = "3306",
         "user" = "root",
         "password" = "123456",
         "database" = "doris_external",
         "table" = "mysql_external_null_1_row");
```
2. export '....'
```
export table test_export_mysql_table (id_varchar,id_datetime,id_tinyint,id_float,id_bigint) TO "oss://dorisdb-data/export/duptbl_cols_exp_oss_1f167fa4_d1a5_11ec_8a7f_00163e0e489a/" PROPERTIES ("column_separator" = "|","line_delimiter" = ",","load_mem_limit" = "10000","timeout" = "360","include_query_id" = "false")  with broker `hdfs_broker` ("fs.oss.accessKeyId" = "xxx", "fs.oss.accessKeySecret" = "xxx", "fs.oss.endpoint" = "xxx");
```
### Expected behavior (Required)
export correctly.
### Real behavior (Required)
Used: 172264, Limit: 10000. Mem usage has exceed the limit of single query
but only one row in the table test_export_mysql_table.
```
mysql> select * from test_export_mysql_table;
+------------+-------------+--------+-----------+----------+-----------+----------------------+---------+---------------------+---------+------------+
| id_tinyint | id_smallint | id_int | id_bigint | id_float | id_double | id_decimal           | id_date | id_datetime         | id_char | id_varchar |
+------------+-------------+--------+-----------+----------+-----------+----------------------+---------+---------------------+---------+------------+
|         13 |        NULL |     33 |      NULL |  1214.95 |      NULL | 1302129092.073558800 | NULL    | 1861-09-12 13:28:18 | NULL    | CANADA     |
+------------+-------------+--------+-----------+----------+-----------+----------------------+---------+---------------------+---------+------------+
1 row in set (0.00 sec)
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - MAIN-RELEASE 8e5b859

### PR DESCRIPTION
**Pull request by caneGuy**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #6003

### CODE REVIEW

File: be/src/connector/mysql_connector.cpp [hunk 1]
```diff
@@ -206,7 +206,7 @@ Status MySQLDataSource::get_next(RuntimeState* state, vectorized::ChunkPtr* chun
         return Status::EndOfFile("finished!");
     }
 
-    *chunk = vectorized::ChunkHelper::new_chunk(*_tuple_desc, state->chunk_size());
+    *chunk = vectorized::ChunkHelper::new_chunk(*_tuple_desc, 0);
```
Comment (imay) [inline be/src/connector/mysql_connector.cpp:209]: I think you don't need to change this value. if you allocate a 0 rows chunk, it can hold 0 rows.

**Review by imay — COMMENTED** [review]
I see that test case, because the default chunk_size in pipeline mode is greater than non-pipeline mode.
So I think it is OK to change the test case.

File: be/src/connector/mysql_connector.cpp (hunk 1 above)
Comment (caneGuy) [reply be/src/connector/mysql_connector.cpp:209]: actually i think fill_chunk will append columns for the chunk. let me test later

Comment (dirtysalt) [reply be/src/connector/mysql_connector.cpp:209]: Consider to do some factor:
1. put `_tuple_desc` into `DataSource`
2. put `create default chunk` into `Data Source`
3. add a method `set default chunk size`in `Data Source`

So all `DataSource` can benefit this code.

File: be/src/connector/connector.h [hunk 2]
```diff
@@ -46,12 +47,16 @@ class DataSource {
         _runtime_filters = runtime_filters;
     }
     void set_read_limit(const uint64_t limit) { _read_limit = limit; }
+    void _init_chunk(vectorized::ChunkPtr* chunk, size_t n) {
```
Comment (dirtysalt) [inline be/src/connector/connector.h:50]: move it as protected method?

File: be/src/connector/connector.h [hunk 3]
```diff
@@ -46,12 +47,16 @@ class DataSource {
...
+        *chunk = vectorized::ChunkHelper::new_chunk(*_tuple_desc, n);
+    }
 
 protected:
     int64_t _read_limit = -1; // no limit
     std::vector<ExprContext*> _conjunct_ctxs;
     const vectorized::RuntimeFilterProbeCollector* _runtime_filters;
     RuntimeProfile* _runtime_profile;
+    TupleDescriptor* _tuple_desc = nullptr;
```
Comment (dirtysalt) [inline be/src/connector/connector.h:59]: const ?

File: be/src/connector/jdbc_connector.h [hunk 4]
```diff
@@ -58,7 +58,6 @@ class JDBCDataSource final : public DataSource {
     ObjectPool _obj_pool;
     ObjectPool* _pool = &_obj_pool;
     RuntimeState* _runtime_state = nullptr;
-    const TupleDescriptor* _tuple_desc = nullptr;
```
[the remark this answers is not in the document]
Comment (caneGuy) [reply be/src/connector/jdbc_connector.h:61]:
```
*chunk = std::make_shared<Chunk>();
    const std::vector<SlotDescriptor*>& slot_descs = _tuple_desc->slots();
    // init column information
    for (auto& slot_desc : slot_descs) {
        ColumnPtr column = ColumnHelper::create_column(slot_desc->type(), slot_desc->is_nullable());
        (*chunk)->append_column(std::move(column), slot_desc->id());
    }
```
for jdbc the chunk was inited in jdbc scanner

**Review by dirtysalt — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Pull request merged**